packages/mermaid/src/dagre-wrapper/index.js

```diff
@@ -54,6 +54,9 @@ const recursiveRender = async (_elem, graph, diagramtype, id, parentCluster, sit
       if (node && node.clusterNode) {
         // const children = graph.children(v);
         log.info('Cluster identified', v, node.width, graph.node(v));
+        // node.graph.setGraph applies the graph configurations such as nodeSpacing to subgraphs as without this the default values would be used
+        // We override the `rankdir` variable to allow for the subgraph's direction to override the graphs direction in the subgraph
+        node.graph.setGraph({ ...graph.graph(), rankdir: node.graph.graph().rankdir });
```

Comment:
I think I've found the issue that was causing the screenshots of the E2E tests to fail.

The issue is that we were updating `marginy` for subgraphs to use their parent value, and that was causing the positioning to shift very slightly.

If we instead only copy over the `ranksep` (i.e. the [`rankSpacing`](https://mermaid.js.org/config/schema-docs/config-defs-flowchart-diagram-config.html#rankspacing)) and the `nodesep` (i.e. the [`nodeSpacing`](https://mermaid.js.org/config/schema-docs/config-defs-flowchart-diagram-config.html#nodespacing)), it seems to work fine.

In the real world, however, it should would slightly modify the subgraphs of anybody that has set `rankSpacing`/`nodeSpacing`, so IMO, this should be in a MINOR release, not a PATCH release.

```suggestion
        const { ranksep, nodesep } = graph.graph();
        node.graph.setGraph({
          ...node.graph.graph(),
          ranksep,
          nodesep,
        });
```